Develop 3 APIs regarding `/booking`

steward379 opened the pull request:
-[FEAT][Database] create `bookings` database. 
-[FEAT][HTML][Style] Create `/booking` page and its CSS, relative files
-[FEAT][Backend] Create 3 Flask APIs regarding `/booking`, which are POST, GET & DELETE
-[FEAT][Frontend] Fetch APIs from backend GET, DELETE in `/booking`
-[FEAT][Frontend] Fetch API from backend POST in `/attraction/<id>`

http://54.253.20.174:3000/
http://54.253.20.174:3000/attraction/8
http://54.253.20.174:3000/booking

steward379 committed:
[fix][css] The mobile layout wrapped.

steward379 committed:
[fix][css] Change booking button to correct width

steward379 committed:
[fix][css] change 'max-width' to 'width' to continue fix booking button

steward379 committed:
[fix][css]change booking button to 'span' and give it some elastic width

steward379 committed:
[fix][css] Change the CSS tolerance larger to booking button.

steward379 committed:
Fulfill form function and add features
-[feat][js] You can now press 'ESC' to close the modal form
-[feat][js] regualr expression for form validation (password,email)
-[fix][html] type=email added to email input
-[fix][css] add line-height to error msg

steward379 committed:
Establish 3 APIs and fetch them  from frontend
[Backend] [FEAT] `api/booking` api established, method `GET`, `POST`, `DELETE`

[Backend] [refactor] connection pool using `dentils.pooled_db`
[Backend] [safety] Database info in .env
[Backend][Frontend][chores] change `Bearer` to none
[Database] [chores] change Mapping to what they are called in the database
[Backend][Environment] make `requirements.txt` worksTO DO:
[Backend][refactor]  Separate every file except this week’s work.

DONE:
[Database][MySQL] Learn how `MySQLdb.cursors.DicCursor` works instead of `tuples`

steward379 committed:
Finish `booking.html` layout design

[css][style] Ensure the background color is grey
[js][date] Ensure Date input is ahead of user's time

steward379 committed:
--the following compiled css fiels

cwpeng reviewed (changes requested):
Show no booking message without checkout form elements if there is no booking data.

steward379 committed:
FEAT: fill in username and email in input value beforehand

- [CRUCIAL][Frontend]Change JavaScript to EventBubbles
So it won't Fetch Multiple Times

steward379 commented:
-- Fix the issue that frustrations bursts when a user forgets to log in but still fill in the form (time, date) in booking form in `attraction.js`(not the payment form), the form data will disappear when he/she logs in afterwards by using `localStorage` to save their form data between he/she sees the login modal and he/she actually logs in.

steward379 committed:
[Fix] Change the booking form to nothing when there's no attraction agenda in the shopping cart

steward379 committed:
[fix] Use Event Bubble to redirect to homepage
[js/css] try to add loading animation

steward379 committed:
[fix][style] Change the style of `Nodata` to be like desgin spec.

cwpeng reviewed (approved):
OK. Done.
